android-components/components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt

```diff
@@ -57,6 +66,26 @@ internal fun Suggestions(
             }
         }
     }
+
+    // The latest list of suggestions with which this composable was recomposed.
+    val currentSuggestions by rememberUpdatedState(suggestions)
+
+    // The latest `onVisibilityStateUpdated` callback with which this composable was composed.
+    val currentOnVisibilityStateUpdated by rememberUpdatedState(onVisibilityStateUpdated)
+
+    LaunchedEffect(Unit) {
+        // This effect is launched on the initial composition. The flow below emits new values when either the latest
+        // list of suggestions changes, or the visibility or position of any suggestion in that list changes.
```

Comment:
Conversely I am actually very happy to see a functional composition block summarized in a comment this way.